jerry-core/ecma/operations/ecma-conversion.c

```diff
@@ -490,27 +490,42 @@ ecma_op_to_string (ecma_value_t value) /**< ecma value */
 } /* ecma_op_to_string */
 
 /**
- * ToPropertyName operation.
+ * ToPropertyKey operation.
+ *
+ * See also:
+ *   ECMA 262 v6, 7.1.14
+ *   ECMA 262 v10, 7.1.14
+ *   ECMA 262 v11, 7.1.19
  *
  * @return NULL - if the conversion fails
  *         ecma-string - otherwise
  */
 ecma_string_t *
-ecma_op_to_prop_name (ecma_value_t value) /**< ecma value */
+ecma_op_to_property_key (ecma_value_t value) /**< ecma value */
 {
-  ecma_check_value_type_is_spec_defined (value);
-
 #if ENABLED (JERRY_ESNEXT)
-  if (ecma_is_value_symbol (value))
+  ecma_value_t key = ecma_op_to_primitive (value, ECMA_PREFERRED_TYPE_STRING);
```

Comment:
ToPrimitve returns with the passed value, if the value is not object. How about keeping the fast path for non-object values?